### ISSUE TITLE
dependency declaration: include pathlib2 conditionally

### ISSUE THREAD

**Issue by linweihs** [fabric/fabric#2180]
fabric package does not declare its dependencies correctly. It unconditionally requires pathlib2.

https://github.com/fabric/fabric/blob/f2ca1093ce1297cbc07fcb750b6f14a872c91656/setup.py#L70

Could we make it including pathlib2 conditionally ?
The last addition to pathlib was in Python 3.7, so with python_version < 3.7 user can include pathlib2 if they want, but with 3.7+ they should not be including it and should use pathlib from standard library.

**Comment by linweihs** [comment]
EDITED

**Comment by davidjmemmett** [comment]
Hi @linweihs, I still don't think I understand what issue you're facing. Are you hitting an incompatibility in a more recent version of Python?

**Comment by linweihs** [comment]
yeah, I have concern of fabric including pathlib2 as dependency. With that said, users are limited to the pathlib2 defined. In stead users should able to decide if they want to include pathlib2 or not in their dependency. 
This actually break in our build system.
